denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc

```diff
@@ -0,0 +1,17 @@
+teleportation_crystal_recipe_item:
+    type: item
+    material: paper
+    display name: <&a>Recipe: Teleportation Crystal
+    mechanisms:
+        nbt: teleportgoesweeeee
+
+teleportation_crystal_recipe_handler:
+    type: world
+    events:
+        on player right clicks with:recipe_item_name:
+        - if !<player.has_flag[teleportation_recipe]>:
+            - flag teleportation_recipe
+            - take scriptname:teleportation_crystal_recipe_item
+        - else
+            - determine cancelled
+            - narrate "You can't use this item again"
```

Comment:
`determine cancelled` will end the queue; you should ideally swap this line and the last line, or specify the `passively` argument.
```suggestion
            - narrate "You can't use this item again"
            - determine cancelled
```